src/Chapter2.hs

```diff
@@ -842,7 +864,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate num l = subList start end (cycle l)
+  where
+    start = num
+    end = num + (length l) - 1
```

Comment:
That's a correct solution! And nice way of reusing `subList` function :+1: 

Though, it does a little bit more than work when the given number `num` is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
I optimized it using `mod` . That way it would not do excessive cycles right?

Reply:
That's right! Nice solution :slightly_smiling_face: